File: frontend/src/types/launchReport.ts
import type { Project } from "./project";

export type LaunchDecision = "ready" | "review" | "not_ready";
export type LaunchChecklistStatus = "done" | "needs_attention" | "todo";

export interface LaunchReportReadiness {
  score: number | null;
  status: string;
  passed: number;
  failed: number;
  unknown: number;
  not_applicable: number;
  total_applicable: number;
  top_gaps: string[];
}

export interface LaunchEvidenceSummary {
  repository_connected: boolean;
  codemap_completed: boolean;
  health_check_healthy: boolean;
  production_url_configured: boolean;
  active_artifacts: number;
  linked_active_artifacts: number;
  unlinked_active_artifacts: number;
  readiness_items_with_linked_artifacts: number;
  readiness_items_without_linked_artifacts: number;
  total_evidence_links: number;
  activity_events: number;
}

export interface ProjectLaunchReport {
  project: Project;
  generated_at: string;
  decision: LaunchDecision;
  headline: string;
  readiness: LaunchReportReadiness;
  evidence_summary: LaunchEvidenceSummary;
  blockers: string[];
  recommended_actions: string[];
}

export interface LaunchChecklistItem {
  key: string;
  label: string;
  status: LaunchChecklistStatus;
  description: string;
  action: string;
  target: string;
}

export interface LaunchChecklistSummary {
  done: number;
  needs_attention: number;
  todo: number;
  total: number;
}

export interface ProjectLaunchChecklist {
  project: Project;
  generated_at: string;
  summary: LaunchChecklistSummary;
  items: LaunchChecklistItem[];
}

File: backend/app/schemas/launch_report.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel

from app.schemas.dashboard import DashboardReadiness
from app.schemas.project import ProjectRead

LaunchDecision = Literal["ready", "review", "not_ready"]
LaunchChecklistStatus = Literal["done", "needs_attention", "todo"]


class LaunchEvidenceSummary(BaseModel):
    repository_connected: bool
    codemap_completed: bool
    health_check_healthy: bool
    production_url_configured: bool
    active_artifacts: int
    linked_active_artifacts: int
    unlinked_active_artifacts: int
    readiness_items_with_linked_artifacts: int
    readiness_items_without_linked_artifacts: int
    total_evidence_links: int
    activity_events: int


class ProjectLaunchReportRead(BaseModel):
    project: ProjectRead
    generated_at: datetime
    decision: LaunchDecision
    headline: str
    readiness: DashboardReadiness
    evidence_summary: LaunchEvidenceSummary
    blockers: list[str]
    recommended_actions: list[str]


class LaunchChecklistItemRead(BaseModel):
    key: str
    label: str
    status: LaunchChecklistStatus
    description: str
    action: str
    target: str


class LaunchChecklistSummary(BaseModel):
    done: int
    needs_attention: int
    todo: int
    total: int


class ProjectLaunchChecklistRead(BaseModel):
    project: ProjectRead
    generated_at: datetime
    summary: LaunchChecklistSummary
    items: list[LaunchChecklistItemRead]
